src/DFA/DFA.fs

```diff
@@ -1,99 +1,59 @@
-            | None -> Partial configuration.RestOfInput
-            | Some nextState -> step (Configuration(nextState, tl))
+            let find state =
+                match Map.tryFind state transitions with
+                | Some(wordMapping: Map<bit list, State seq>) -> Map.tryFind hd wordMapping
+                | _ -> None
+                |> (Option.toList)
+                |> List.toSeq
+                |> Seq.concat in
+
+            states |> Seq.map (find) |> Seq.concat |> Seq.distinct |> recognize tl
 
     member this.StartState = startState
-    member this.Recognize str = step (Configuration(startState, str))
+    member this.FinalStates = finalStates
+    member this.Transitions = transitions
+
+    member this.Recognize input =
+        recognize input <| Seq.singleton startState
 
     member this.ToDot() =
         let nodes = ResizeArray<_>()
         let edges = ResizeArray<_>()
-        let mutable firstFreeNodeId = 0
-
-        let dfs (state: State<_>) =
-            let toProcess = Stack<State<_> * (int -> unit)>[(state, (fun _ -> ()))]
-            let copyOfState = Dictionary<State<_>, int>()
-
-            while toProcess.Count > 0 do
-                let state, addTransitions = toProcess.Pop()
-                let exists, newState = copyOfState.TryGetValue state
-
-                if not exists then
-                    let curNodeId = firstFreeNodeId
-                    firstFreeNodeId <- firstFreeNodeId + 1
-                    let nodeColor = if state.IsStart then "green" else "white"
-                    let nodeShape = if state.IsFinal then "doublecircle" else "circle"
 
-                    nodes.Add(
-                        $"{curNodeId} [fillcolor={nodeColor} shape={nodeShape} style=filled label=\"{state.Id}\"]"
-                    )
+        for (fromState, letterInfo) in Map.toSeq transitions do
+            let nodeColor = if fromState = startState then "green" else "white"
 
-                    addTransitions curNodeId
-
-                    for (char, targetState) in state.GetAllTransitions() do
-                        toProcess.Push(
-                            targetState,
-                            (fun newTargetStateId -> edges.Add($"{curNodeId} -> {newTargetStateId} [label=\"{char}\"]"))
-                        )
-
-                    copyOfState.Add(state, curNodeId)
+            let nodeShape =
+                if Seq.contains fromState finalStates then
+                    "doublecircle"
                 else
-                    addTransitions newState
+                    "circle"
+
+            nodes.Add(
+                $"{fromState.Id} [fillcolor={nodeColor} shape={nodeShape} style=filled label=\"{fromState.Name}\"]"
+            )
 
-        dfs this.StartState
+            for (letter, toStates) in Map.toSeq letterInfo do
+                edges.Add($"{fromState.Id} -> %A{toStates |> Seq.distinct} [label=\"%A{letter}\"]")
```

Comment:
```suggestion
            for (letter, toStates) in Map.toSeq letterInfo do
                for toState in toStates do
                    edges.Add($"{fromState.Id} -> %A{toState.Id} [label=\"%A{letter}\"]")
```
This will probably fix dot output